proto/public.proto

```diff
@@ -155,6 +155,16 @@ message Scheme {
   }
 }
 
+message Headers {
+  message Header {
+    string name = 1;
+    string value = 2;
```

Comment:
Probably good idea to have a comment here explaining that only one of these fields will be set, depending on if the header value is UTF-8 decodable and also explaining why we didn't just use a oneof.

Reply:
Good point! I added a clarifying comment.